app/controllers/service_controller.rb

```diff
@@ -150,6 +157,22 @@ def service_form_fields
     }
   end
 
+  def generic_object
+    return unless init_show_variables
+
+    id = params[:show] || params[:x_show]
+    if id.present?
+      @lastaction = "generic_object"
+      @item = @record.generic_objects.find(from_cid(id)).first
+      drop_breadcrumb(:name => _("%{name} (All Generic Objects)") % {:name => @record.name},
+                      :url  => show_link(@record, :display => @display))
+      drop_breadcrumb(:name => @item.name, :url => "/#{controller_name}/show/#{@record.id}?display=generic_objects/show=#{id}")
```

Comment:
We really should use methods for generating URLs. At least in new code. I we don't then things like restful routes will bite us again and again.